redis/server/install.sls

```diff
@@ -0,0 +1,104 @@
+include:
+  - redis.common
+
+{% from "redis/map.jinja" import redis_settings as r with context %}
+
+    {%- if r.install_from == 'source' %}
+
+redis_user_group:
+  group.present:
+    - name: {{ r.group }}g
+  user.present:
+    - name: {{ r.user }}g
+    - gid_from_name: True
+    - home: {{ r.home }}
+    - require:
+      - group: redis_group
+  file.managed:
+    - name: /etc/init/redis-server.conf
+    - template: jinja
+    - source: salt://redis/files/upstart.conf.jinja
+    - mode: '0750'
+    - user: root
+    - group: root
+    - context:
+        conf: /etc/redis/redis.conf
+        user: {{ r.user }}g
+        bin: {{ r.bin }}
+    - require:
+      - sls: redis.common
```

Comment:
I'm not a huge fan of multiple states under the same ID. It can't be decomposed at call time.

Reply:
Okay, It's a valid comment. I tend to decompose to SLS level, not State ID level, and follow that design. Also, Travis CI/CD operates at the SLS level across most formulae.  But I will split this as suggested.